src/core/main/Configuration/ConfigurationGetExtensions.cs

```diff
@@ -0,0 +1,171 @@
+            {
+                return defaultValue;
+            }
+
+            return ConvertOrDefault(value, defaultValue);
+        }
+
+        /// <summary>
+        /// Get a configuration value that can be behind multiple keys - or default.
+        /// The keys are checked _in order_.
+        ///
+        /// This is a convenience overload for Get(string[] keys, T defaultValue).
+        /// </summary>
+        public static T Get<T>(this IConfiguration config, string keyPrimary, string keySecondary, T defaultValue)
+        {
+            return config.Get(new[] {keyPrimary, keySecondary}, defaultValue);
+        }
+        
+        /// <summary>
+        /// Get a configuration value that can be behind multiple keys - or default.
+        /// The keys are checked _in order_.
+        ///
+        /// This is a convenience overload for Get(string[] keys, T defaultValue).
+        /// </summary>
+        public static T Get<T>(this IConfiguration config, string keyPrimary, string keySecondary, string keyTertiary, T defaultValue)
+        {
+            return config.Get(new[] {keyPrimary, keySecondary, keyTertiary}, defaultValue);
+        }
+
+        /// <summary>
+        /// Get a configuration value that can be behind multiple keys - or default.
+        /// The keys are checked _in order_.
+        ///
+        /// The idea behind this, is that you might have a scenario where you have multiple
+        /// specific things that might share a common config value.
+        ///
+        /// Example:
+        /// Multiple databases in code, but actually running on the same host. In this case, you
+        /// probably want to avoid having to duplicate the hostname to every single database config,
+        /// but you want to retain the option of each of them overriding the "shared" value.
+        ///
+        /// Instead of the following, where there is no sharing...
+        /// {
+        ///   "db_a": { "host": "shared_host", "name": "a" },
+        ///   "db_b": { "host": "shared_host", "name": "b" }
+        /// }
+        ///
+        /// You could now have...
+        ///
+        /// {
+        ///   "db_shared": { "host": "shared_host" },
+        ///   "db_a": { "name": "a" },
+        ///   "db_b": { "name": "b" }
+        /// }
+        ///
+        /// And then use:
+        /// config.Get("db_a:host", "db_shared:host", "defaultValue");
+        /// </summary>
+        public static T Get<T>(this IConfiguration config, IEnumerable<string> keys, T defaultValue)
+        {
+            foreach (var key in keys)
+            {
+                var value = config[key];
+
+                if (!string.IsNullOrEmpty(value))
+                {
+                    return ConvertOrDefault(value, defaultValue);
+                }
+            }
+
+            return defaultValue;
+        }
+
+        /// <summary>
+        /// This is a convenience overload for GetFromCommaSeparatedList(string[] keys, T defaultValue)
```

Comment:
**nitpick**: The default value is type `List<T>`

This is also the case for the next two extension methods.

Reply:
That should be fixed now.